lib/bamboo/sent_email.ex

```diff
@@ -89,7 +89,7 @@ defmodule Bamboo.SentEmail do
 
   defp do_get(id) do
     Enum.find all, nil, fn(email) ->
-      get_id(email) == id
+      get_id(email) == String.downcase(id)
```

Comment:
I'm not sure how common that is but I think that's ok to keep it then. I think it would be good to add a test case where the passed in ID param is uppercase. That way we can ensure it doesn't break in the future